portfolio/src/main/webapp/quiz.js

```diff
@@ -2,40 +2,88 @@
+		['San Francisco, CA', 'Vancouver, BC', 'Toronto, ON', 'Palo Alto CA'], 1));
+questions.push(
+    new Question('Where do I attend university?',
+		['University of British Columbia', 'Simon Fraser University',
+		  'Stanford University', 'New York University'], 0));
+questions.push(
+    new Question('In which of these classes did I receive the highest grade?',
+		['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], 1));
+questions.push(
+    new Question('In which of these classes did I receive the lowest grade?',
+		['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], 3));
+
 function presentQuiz() {
-  // create array of quiz questions 
-  const questions;
+  const questionsContainer = document.getElementById('questions-container');
+  for (let questionNum = 0; questionNum < questions.length; questionNum++) {
+    const q = questions[questionNum];
+    const answers = q.getAnswers();
+    const correctIndex = q.getCorrectAnswerIndex();
+    // Display the options
+    const questionStatement = document.createElement('p');
+    const question = document.createTextNode(`Question ${questionNum}: ${q.getQuestion()}`);
+    questionStatement.appendChild(question);
 
-  const quizContainer = document.getElementById('quiz-container');
-  const questionContainer = document.getElementById('question-container');
-  const answersContainer = document.getElementById('answers-container');
-  const submitContainer = document.getElementById('submit-container');
+    const quizArea = document.createElement('select');
+    quizArea.setAttribute('id', `question-${questionNum}`);
+    for (let answerIndex = 0; answerIndex < answers.length; answerIndex++) {
+      const option = document.createElement('option');
+      option.setAttribute('value', answerIndex);
+      const answer = document.createTextNode(answers[answerIndex]);
+      option.appendChild(answer);
+      quizArea.appendChild(option);
+    }
 
-  for (q in questions) {
-    // display the question in a container
-    questionContainer.innerText = q.getQuestion();
-    // Display the options
-    // Use radio buttons
-    // Have a submit button
-    // When user clicks submit button display correct answer
+    const emptyAnswer = document.createElement('option');
+    emptyAnswer.setAttribute('selected', '');
+    emptyAnswer.setAttribute('value', '');
+    quizArea.appendChild(emptyAnswer);
+
+    questionsContainer.appendChild(questionStatement);
+    questionsContainer.appendChild(quizArea);
+  }
+}
+
+function checkSubmission() {
+  const submitContainer = document.getElementById('submit-container');
+  submitContainer.innerText = '';
+  for (let questionNum = 0; questionNum < questions.length; questionNum++) {
+    const q = questions[questionNum];
+    const answers = q.getAnswers();
```

Comment:
nit - i don't think answers is used.

Reply:
That's something I didn't catch, I'll remove that.